homeassistant/components/matrix/__init__.py

```diff
@@ -34,23 +50,37 @@
 CONF_WORD = "word"
 CONF_EXPRESSION = "expression"
 
+EVENT_MATRIX_COMMAND = "matrix_command"
+
 DEFAULT_CONTENT_TYPE = "application/octet-stream"
 
 MESSAGE_FORMATS = [FORMAT_HTML, FORMAT_TEXT]
 DEFAULT_MESSAGE_FORMAT = FORMAT_TEXT
 
-EVENT_MATRIX_COMMAND = "matrix_command"
-
 ATTR_FORMAT = "format"  # optional message format
```

Comment:
Hi @PaarthShah , really great work you have done so far. I came to the HA repo to start a PR introducing the  message type [m.notice](https://spec.matrix.org/v1.3/client-server-api/#mroommessage-msgtypes) to the existing matrix component, when I saw yours.

Could you imagine introducing a new optional attribute `ATTR_TYPE = "type"` which can hold a value `notice` and passes it on to the send function, just like the `ATTR_FORMAT` does?

The motivation for this comes from having a chatbot to interact with HA. When I write some kind of command, I want to get an immediate response as acknowledgement but don't want all my other matrix clients to fire a notification sound when this bot response arrives.

Reply:
Hi @efelon! Appreciate you taking the time to look at this!

Totally agreed that implementing the use of `m.notice` would be a good and important addition. I'd be happy to take a crack at it (and fix that last test failure 🙃) in the upcoming week!

My only _slight_ concern is that I didn't originally want to do extra feature work in this initial PR since I didn't want to confound the 1:1 translation of the old code to the new code... But honestly adding something this small and fundamental isn't going to delay its review/merging any further, so I'm in

Reply:
In truth, my very original motivation for this PR was that I wanted HASS to be able to participate in/initiate entire Threads/respond to reactions! Heck, we even have live location sharing in Element/synapse now...

My long-term goal is to let this integration be fully capable of sending/responding to any particular matrix event

Reply:
> But honestly adding something this small and fundamental isn't going to delay its review/merging any further,

Exactly what I thought. Since it is only a few lines in the end, I decided to leave a comment and not to dive into the overhead of creating a PR only for it.

> , so I'm in

🤩

> My long-term goal is to let this integration be fully capable of sending/responding to any particular matrix event

That is one kind of a vision, with a lot of work behind. I will keep an eye on your work and wish you all the best for it. Keep in mind that the matrix spec. is still very fluid. Especially in terms of new features like threads or live location sharing. IMHO the next important thing is having the E2EE handling (cross signing, verification requests, ...) somehow manageable in HA (in the config flow I guess).